app/models/redirection.rb

```diff
@@ -0,0 +1,16 @@
+class Redirection < ActiveRecord::Base
+  belongs_to :next, class_name: "Redirection"
+  has_one :previous, class_name: "Redirection", foreign_key: "next_id"
+
+  validates :next_id, uniqueness: true, allow_blank: true
```

Comment:
Specifically: Is this to allow gabe -> edward (with edward pointing nowhere) instead of gabe -> edward -> gabe? I think that breaks the ring, right?

(Maybe there's another reason and I'm totally misreading this.)